src/main/java/org/jboss/common/beans/property/finder/PropertyEditorFinder.java

```diff
@@ -86,22 +80,26 @@ private static void init() {
     }
 
     /**
-     * Returns instance of property finder. Instance is loaded as {@java.util.ServiceLoader} service.
+     * Returns instance of property finder. Instance is loaded as {@link java.util.ServiceLoader} service.
      * If no service is found, this method, this method will return <b>null</b>.
      *
      * @return Instance of PropertyEditorFinder.
      */
-    public static PropertyEditorFinder getInstance() {
+    public static PropertyEditorFinder getInstance(ClassLoader classLoader) {
         PropertyEditorFinder result = _INSTANCE;
         if (result == null) {
             synchronized (_LOCK) {
                 result = _INSTANCE;
                 if (result == null) {
-                    init();
+                    init(classLoader);
                     result = _INSTANCE;
                 }
             }
         }
         return result;
     }
+
+    public static PropertyEditorFinder getInstance() {
+        return getInstance(PropertyEditorFinder.class.getClassLoader());
```

Comment:
To be compatible this needs to use a doPrivileged call to get the TCCL and then pass that. Per the ServiceLoader javadoc, the previous call at L81 resulted in use of the TCCL, not the calling class' classloader.